### PR TITLE
Thread: update timeout when reruning WaitSynch

### LINKED ISSUE #1307: Waiting on a WaitSynchronization with timeout will reset the timeout each time the thread is awakened
If a call to WaitSynchronization1/N with timeout causes a thread to sleep, the timeout will be reset every time we repeat the WaitSynchronization call if it sends the thread to sleep again.

We should keep track of the remaining time in the Wakeup event and adjust the timeout parameter accordingly for the next call (And then restore it).

### PR DESCRIPTION
**Pull request by wwylele**
Fix #1307 
Here is a [small test](https://github.com/wwylele/ctrhwtest/tree/master/waitsync_timeout_test) for this.

### CODE REVIEW

File: src/core/hle/kernel/thread.cpp [hunk 1]
```diff
@@ -219,6 +219,36 @@ static void SwitchContext(Thread* new_thread) {
 
             // SVC instruction is 2 bytes for THUMB, 4 bytes for ARM
             new_thread->context.pc -= thumb_mode ? 2 : 4;
+
+            // Get the SVC call instruction
+            u16 thumb_inst = Memory::Read16(new_thread->context.pc & 0xFFFFFFFE);
```
Comment (lioncash) [inline src/core/hle/kernel/thread.cpp]: This should be its own function that returns a pair or a tuple containing the low or high addresses.

File: src/core/hle/kernel/thread.cpp [hunk 2]
```diff
@@ -219,6 +219,36 @@ static void SwitchContext(Thread* new_thread) {
...
+            } else if ((thumb_mode && thumb_inst == 0xDF25) || (!thumb_mode && inst == 0x0F000025)) {
+                // svc #0x25 (WaitSynchronizationN)
+                reg_timeout_low = &new_thread->context.cpu_registers[0];
+                reg_timeout_high = &new_thread->context.cpu_registers[4];
+            } else {
+                UNREACHABLE();
+            }
+
+            // Update the timeout parameter
+            s64 timeout = ((s64)*reg_timeout_high << 32) | *reg_timeout_low;
```
Comment (lioncash) [inline src/core/hle/kernel/thread.cpp]: This should also be its own function that performs the update.

**Comment by wwylele** [comment]
Addressed

**Comment by Subv** [comment]
LGTM, but this seems more of a band-aid solution until we change how our scheduler works.

**Comment by bunnei** [comment]
LGTM

**Pull request merged**